items/src/test/scala/weco/api/items/services/SierraItemUpdaterTest.scala

```diff
@@ -0,0 +1,503 @@
+    method = CatalogueAccessMethod.NotRequestable,
+    status = Some(CatalogueAccessStatus.TemporarilyUnavailable),
+    note = Some(
+      "Item is in use by another reader. Please ask at Library Enquiry Desk."
+    ),
+    terms = None
+  )
+
+  val onlineRequestAccessCondition = DisplayAccessCondition(
+    method = CatalogueAccessMethod.OnlineRequest,
+    status = CatalogueAccessStatus.Open
+  )
+
+  val availableDates = Some(
+    List(
+      AvailabilitySlot(
+        "2024-04-25T09:00:00.000Z",
+        "2024-04-25T19:00:00.000Z"
+      ),
+      AvailabilitySlot(
+        "2024-04-26T09:00:00.000Z",
+        "2024-04-26T17:00:00.000Z"
+      )
+    )
+  )
+
+  class DummyItemUpdater(
+    itemTransform: Seq[DisplayItem] => Seq[DisplayItem] = identity
+  ) extends ItemUpdater {
+    override val identifierType: DisplayIdentifierType =
+      DisplayIdentifierType.SierraSystemNumber
+
+    override def updateItems(
+      items: Seq[DisplayItem]
+    ): Future[Seq[DisplayItem]] = Future {
+      itemTransform(items)
+    }
+  }
+
+  describe("with SierraItemUpdater") {
+    val workWithUnavailableItemNumber = createSierraItemNumber
+    val workWithAvailableItemNumber = createSierraItemNumber
+
+    val workWithAvailableItem = CatalogueWork(
+      id = createCanonicalId,
+      title = None,
+      identifiers = Nil,
+      items = List(
+        availableItem(workWithAvailableItemNumber)
+      )
+    )
+
+    val itemStates = Table(
+      (
+        "Sierra Responses",
+        "Content api Venue Responses",
+        "Catalogue Work",
+        "AccessCondition",
+        "AvailableDates"
+      ),
+      (
+        onHoldItemResponses(workWithAvailableItemNumber),
+        Nil,
+        List(availableItem(workWithAvailableItemNumber)),
+        onHoldAccessCondition,
+        None
+      ),
+      (
+        onHoldItemResponses(workWithUnavailableItemNumber),
+        Nil,
+        List(
+          temporarilyUnavailableItem(workWithUnavailableItemNumber)
+        ),
+        onHoldAccessCondition,
+        None
+      ),
+      (
+        availableItemResponses(workWithAvailableItemNumber),
+        Seq((contentApiVenueRequest("library"), contentApiVenueResponse())),
+        List(availableItem(workWithAvailableItemNumber)),
+        onlineRequestAccessCondition,
+        availableDates
+      ),
+      (
+        availableItemResponses(workWithUnavailableItemNumber),
+        Seq((contentApiVenueRequest("library"), contentApiVenueResponse())),
+        List(
+          temporarilyUnavailableItem(workWithUnavailableItemNumber)
+        ),
+        onlineRequestAccessCondition,
+        availableDates
+      ),
+      (
+        missingItemResponse(workWithAvailableItemNumber),
+        Seq((contentApiVenueRequest("library"), contentApiVenueResponse())),
+        List(availableItem(workWithAvailableItemNumber)),
+        availableOnlineAccessCondition,
+        availableDates
+      )
+    )
```

Comment:
Should we have some test cases for deepstore so we test the logic in `deepstoreItemAvailabilities`?